ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/translator/ASTConverter.java

```diff
@@ -788,6 +789,9 @@ public ASTNode visitCall(RexCall call) {
           }
         }
         break;
+      case IS_DISTINCT_FROM:
+        // convert IS DISTINCT FROM to NOT (IS NOT DISTINCT FROM)
+        return visitCall((RexCall) RexUtil.not(rexBuilder.makeCall(SqlStdOperatorTable.IS_NOT_DISTINCT_FROM, call.getOperands())));
```

Comment:
If you rely on `buildAST(SqlOperator op, List<ASTNode> children)`, you would not have to declare the function in `FunctionRegistry`?
i) Call `visitCall` on the call operands, ii) then `buildAST` with `SqlStdOperatorTable.IS_NOT_DISTINCT_FROM`, iii) then `buildAST` with the output of the previous one.

Reply:
Done!

Reply:
Do you still need to add `is_not_distinct_from` in FunctionRegistry.java after this change? The reason why I am asking is that adding it to FunctionRegistry will make it user facing, hopefully we can avoid that.